src/EditorFeatures/Test/Diagnostics/TestDiagnosticAnalyzerService.cs

```diff
@@ -68,6 +68,14 @@ public override string FullPath
                 }
             }
 
+            public override string Id
+            {
+                get
+                {
+                    return nameof(TestAnalyzerReferenceByLanguage);
```

Comment:
Will this be unique? We should probably define a protected method `CreateUniqueId` or some such on AnalyzerReference and use it for all implementations that just want a default implementation for `Id`?